internal/mage/engine.go

```diff
@@ -287,44 +294,92 @@ func (t Engine) Dev(ctx context.Context) error {
 }
 
 func devEngineContainer(c *dagger.Client, arches []string) []*dagger.Container {
-	buildkitRepo := c.Git(buildkitRepo, dagger.GitOpts{KeepGitDir: true}).Branch(buildkitBranch).Tree()
-
 	platformVariants := make([]*dagger.Container, 0, len(arches))
 	for _, arch := range arches {
-		buildkitBase := c.Container(dagger.ContainerOpts{
-			Platform: dagger.Platform("linux/" + arch),
-		}).Build(buildkitRepo)
-
-		// build the shim binary
-		shimBin := util.GoBase(c).
-			WithEnvVariable("GOOS", "linux").
-			WithEnvVariable("GOARCH", arch).
-			WithExec([]string{
-				"go", "build",
-				"-o", "./bin/" + shimBinName,
-				"-ldflags", "-s -w",
-				"/app/cmd/shim",
+		platformVariants = append(platformVariants, c.
+			Container(dagger.ContainerOpts{Platform: dagger.Platform("linux/" + arch)}).
+			From("alpine:"+alpineVersion).
+			WithExec([]string{"apk", "add", "git", "openssh", "pigz", "xz"}).
```

Comment:
Ooh, are we completely decoupled from the upstream image now, so we need to add these ourselves for e.g. the Git source? Interesting. (Not a bad thing, just making a mental note.)

Reply:
Yeah we no longer want to spend time building the buildkitd binary from upstream and the option of trying to still use their dockerfile but from different targets felt too messy to me. IMO it's much easier to follow in the Go SDK than in the Dockerfile syntax :-)